### PR TITLE
Error handling auth

### PR DESCRIPTION
**Pull request by koceljevac**

### CODE REVIEW

File: lib/core/di/injection.dart [hunk 1]
```diff
@@ -45,4 +47,9 @@ abstract class InjectableModule {
   SpreadsheetDownloader provideApiClient(Dio dio) {
     return SpreadsheetDownloader(dio);
   }
+  @singleton
+  StreamController<void> get navigationStreamController {
```
Comment (zeljkons1992) [inline lib/core/di/injection.dart:51]: Delete this

File: lib/core/domain/exceptions/auth_exception_message.dart [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class AuthExceptionMessage implements Exception {
+  String cause;
+  AuthExceptionMessage(this.cause);
+
+  @override
+  String toString() => cause;
```
Comment (zeljkons1992) [inline lib/core/domain/exceptions/auth_exception_message.dart:6]: What do you use this for?

Comment (zeljkons1992) [reply lib/core/domain/exceptions/auth_exception_message.dart:6]: This class?

File: lib/features/auth/data/repository/auth_repository_impl.dart [hunk 3]
```diff
@@ -12,11 +16,23 @@ class AuthRepositoryImpl implements AuthRepository {
...
+    } on AuthExceptionMessage catch (e) {
+      return DataState.error(e.cause);
+    } on PlatformException catch (e) {
+      if (e.code == 'network_error') {
+        return DataState.error("no_network");
+      } else {
+        return DataState.error(e.toString());
+      }
+    } catch (e) {
+      return DataState.error("User canceled $e");
```
Comment (zeljkons1992) [inline lib/features/auth/data/repository/auth_repository_impl.dart:35]: Add a generic error message, something like "Technical problems occurred, please try again later"

File: lib/features/auth/presentaion/screens/sign_in_screen.dart [hunk 4]
```diff
@@ -7,27 +9,58 @@ import 'package:vegawallet/features/auth/presentaion/bloc/auth/auth_bloc.dart';
...
-                  duration: Duration(seconds: 3),
+                SnackBar(
+                  content: Text(translate(state.message, localization)),
+                  duration: const Duration(seconds: 3),
                 ),
               );
-            case AuthLoginWithGoogleSuccess _:
-              return context.go("/");
+            // case AuthLoginWithGoogleSuccess _:
+            //   return context.go("/");
```
Comment (zeljkons1992) [inline lib/features/auth/presentaion/screens/sign_in_screen.dart:62]: Delete this

File: lib/core/domain/exceptions/auth_exception_message.dart (hunk 2 above)
Comment (koceljevac) [reply lib/core/domain/exceptions/auth_exception_message.dart:6]: Exception handler

**Review by VojinBjelica — APPROVED** [review]

**Review by zeljkons1992 — DISMISSED** [review]

**Review by VojinBjelica — APPROVED** [review]

**Pull request merged**